src/plots/cartesian/graph_interact.js

```diff
@@ -624,10 +624,14 @@ function hover(gd, evt, subplot) {
     if(!hoverChanged(gd, evt, oldhoverdata)) return;
 
     if(oldhoverdata) {
-        gd.emit('plotly_unhover', { points: oldhoverdata });
+        gd.emit('plotly_unhover', {
+            event: evt,
+            points: oldhoverdata
+        });
     }
 
     gd.emit('plotly_hover', {
+        event: evt,
```

Comment:
Great. This will work for all subplots that call `Fx.hover` if I remember correctly that means: cartesian, ternary, geo and mapbox subplots. @n-riesco would you mind double checking?

Now, at the very least, we should make sure `event` is passed in pie chart hover data. Pie hover is a little [different](https://github.com/plotly/plotly.js/blob/388287b09dd88634b9603a8599321ad73d95c352/src/traces/pie/plot.js#L116) but shouldn't make it too hard to implement.

This leaves our gl2d, gl3d and `parcoords` (and polar but who cares about polar?). I'll be ok leaving them out of the first iteration. We should instead spent some time down the road merging their hover logic with `Fx.hover`.